stdlib/public/Concurrency/TaskStatus.cpp

```diff
@@ -256,11 +256,14 @@ static bool withStatusRecordLock(AsyncTask *task,
 /*************************** RECORD MANAGEMENT ****************************/
 /**************************************************************************/
 
+SWIFT_EXPORT_FROM(swift_Concurrency)
 SWIFT_CC(swift)
-static bool swift_task_addStatusRecordImpl(TaskStatusRecord *newRecord) {
-  auto task = swift_task_getCurrent();
+bool swift_task_addStatusRecordWithChecks(
+    TaskStatusRecord *newRecord,
+    llvm::function_ref<bool(ActiveTaskStatus status)> shouldAddRecord) {
```

Comment:
Implementations of functions from interfaces should be scope-qualified so that we get redeclaration checking.  That is, `bool swift::swift_task_addStatusRecordWithChecks` (but renamed as discussed above).

The `fooImpl` functions don't do that because they're not directly implementing functions from interfaces.  But we do still get signature checking for them, since the direct implementation of `swift::foo` calls the `fooImpl` stub, which lacks any other forward declaration.